core/src/main/java/org/elasticsearch/index/query/BoostingQueryBuilder.java

```diff
@@ -44,9 +44,9 @@
 
     public static final String NAME = "boosting";
 
-    private QueryBuilder positiveQuery;
+    private QueryBuilder positiveQuery = EmptyQueryBuilder.PROTOTYPE;
 
-    private QueryBuilder negativeQuery;
+    private QueryBuilder negativeQuery = EmptyQueryBuilder.PROTOTYPE;
```

Comment:
I see how having a null default might lead to needing null checks in doXContent etc. though, let's discuss this. I think we have the opposite problem in other queries, will comment below.